### PR TITLE
combine ca secrets

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Combine ca secrets

**Release note**:
```release-note
Combine the two existing CA secrets into a single one
```

### CODE REVIEW

File: api/pkg/resources/controllermanager/deployment.go [hunk 1]
```diff
@@ -114,13 +114,8 @@ func Deployment(data *resources.TemplateData, existing *appsv1.Deployment) (*app
...
 		{
-			Name:      resources.CACertSecretName,
-			MountPath: "/etc/kubernetes/ca-cert",
-			ReadOnly:  true,
-		},
-		{
-			Name:      resources.CAKeySecretName,
-			MountPath: "/etc/kubernetes/ca-key",
+			Name:      resources.CASecretName,
+			MountPath: "/etc/kubernetes/ca",
```
Comment (alvaroaleman) [inline api/pkg/resources/controllermanager/deployment.go]: What about using `/etc/kubernetes/pki` here like kubeadm does?

**Review by alvaroaleman — CHANGES_REQUESTED** [review]
Please use `/etc/kubernetes/pki` instead of `/etc/kubernetes/ca` as mountPath in the various components, other than that LGTM

File: api/pkg/resources/controllermanager/deployment.go (hunk 1 above)
Comment (thz) [reply api/pkg/resources/controllermanager/deployment.go]: Wouldn't `/etc/kubernetes/pki/ca` be better? Where would we mount other keys like client-certs?

Comment (alvaroaleman) [reply api/pkg/resources/controllermanager/deployment.go]: Good point, and good suggestion

Comment (mrIncompetent) [reply api/pkg/resources/controllermanager/deployment.go]: Done

**Comment by mrIncompetent** [comment]
@alvaroaleman PTAL

File: api/pkg/resources/etcd/statefulset.go [hunk 2]
```diff
@@ -157,7 +157,7 @@ func StatefulSet(data *resources.TemplateData, existing *appsv1.StatefulSet) (*a
 					MountPath: "/etc/etcd/tls",
 				},
 				{
-					Name:      resources.CACertSecretName,
```
Comment (alvaroaleman) [inline api/pkg/resources/etcd/statefulset.go]: This one is not updated, would be nice if we had it consistent

**Review by alvaroaleman — CHANGES_REQUESTED** [review]

**Review by alvaroaleman — DISMISSED** [review]

**Review by alvaroaleman — APPROVED** [review]

**Pull request merged**